pkg/backporter/httpcache.go

```diff
@@ -0,0 +1,131 @@
+package backporter
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"sync"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+)
+
+type bugzillaCache struct {
+	lock  sync.Mutex
+	cache map[string][]byte
+}
+
+func (bc *bugzillaCache) Get(key string) ([]byte, bool) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	cachedVal, ok := bc.cache[key]
+	return cachedVal, ok
+}
+
+func (bc *bugzillaCache) Set(key string, respBytes []byte) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	bc.cache[key] = respBytes
+}
+
+func (bc *bugzillaCache) Delete(key string) {
+	bc.lock.Lock()
+	defer bc.lock.Unlock()
+	delete(bc.cache, key)
+}
+func newBugzillaCache() *bugzillaCache {
+	return &bugzillaCache{cache: map[string][]byte{}}
+}
+
+// CachedTransport is an implementation http.RoundTripper
+// which first checks for cached values
+type CachedTransport struct {
+	mirror    *bugzillaCache
+	Transport http.RoundTripper
+}
+
+// RoundTrip will first check if there are any cached responses and return that
+// if not it will make an HTTP call using the DefaultTransport
+func (t *CachedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Method != "GET" {
+		return http.DefaultTransport.RoundTrip(req)
```

Comment:
In order to make this testable it makes sense to make the http roundtripper a property of `CachedTransport` rather than hardcoding it to the `http.DefaultTransport` (and default it to `http.DefaultTransport` in the constructor)